File: frontend/src/app/services/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { Employee } from '../model/employee';
import {environment} from "../../environments/environment";

@Injectable({
    providedIn: 'root'
})
export class EmployeeService {
    private employeesUrl = `${environment.apiUrl}/employees`;

    constructor(private http: HttpClient) {}

    // Fetch all employees
    getEmployees(): Observable<Employee[]> {
        return this.http.get<Employee[]>(this.employeesUrl);
    }

    // Save a new employee
    saveEmployee(employee: Employee): Observable<Employee> {
        return this.http.post<Employee>(this.employeesUrl, employee);
    }

    // Update an existing employee
    updateEmployee(id: number, employee: Employee): Observable<Employee> {
        const url = `${this.employeesUrl}/${id}`;
        return this.http.put<Employee>(url, employee);
    }

    // Delete an employee
    deleteEmployee(id: number): Observable<void> {
        const url = `${this.employeesUrl}/${id}`;
        return this.http.delete<void>(url);
    }

    // Update an employee's password
    updatePassword(id: number, password: string): Observable<void> {
        const url = `${this.employeesUrl}/${id}/password`;
        return this.http.put<void>(url, { password });
    }
}

File: backend/etams/src/main/java/com/brubaker/etams/service/EmployeeService.java
package com.brubaker.etams.service;

import com.brubaker.etams.dto.EmployeeDTO;

import java.util.List;

public interface EmployeeService {
    // Get all employees with task counts
    List<EmployeeDTO> getAllEmployeesWithTaskCount();

    // Create a new employee
    EmployeeDTO createEmployee(EmployeeDTO employeeDTO);

    // Update an existing employee
    EmployeeDTO updateEmployee(Integer id, EmployeeDTO employeeDTO);

    // Update the password for an employee
    void updatePassword(Integer id, String password);

    // Optional: Delete an employee (if needed)
    void deleteEmployee(Integer id);
}
